fabfile.py

```diff
@@ -1,2435 +1 @@
-
-
-def get_upload_dir(path=None):
-    path = path or env.get('upload_root', 'var/uploads')
-    if path != '/':
-        path = join(env.projectpath, path)
-    return path
-
-
-@task
-def database_download():
-    """
-    Dumps and downloads the database from the target server
-    """
-    destination = join('./', get_db_dump_name())
-    if is_link(destination):
-        print('Clearing symlink at %s to make way for downloaded file' % (destination))
-        local('rm %s' % (destination))
-    execute(database_dump)
-    get(remote_db_path(), destination)
-    remote_path = get_upload_dir()
-    rsync_path = "%s@%s:%s" % (env.user, env.host_string, remote_path)
-    local_venv = env.get("local_venv", "./venv")
-    with settings(host_string="localhost", venvpath=local_venv,
-                  user=getuser(), projectpath=os.getcwd()):
-        # TODO: I should check the local upload_path. But in practice
-        # it's a developer's machine, probably uses standard.
-        local_path = get_upload_dir('var/uploads')
-        run("rsync -a %s/ %s" % (rsync_path, local_path))
-
-
-@task
-def database_upload():
-    """
-    Uploads a local database backup to the target environment's server
-    """
-    if(env.wsginame != 'dev.wsgi'):
-        put(get_db_dump_name(), remote_db_path())
-        remote_path = get_upload_dir()
-        rsync_path = "%s@%s:%s/" % (env.user, env.host_string, remote_path)
-        local_venv = env.get("local_venv", "./venv")
-        with settings(host_string="localhost", venvpath=local_venv,
-                      user=getuser(), projectpath=os.getcwd()):
-            # TODO: I should check the local upload_path. But in practice
-            # it's a developer's machine, probably uses standard.
-            local_path = get_upload_dir('var/uploads')
-            run("rsync -a %s/ %s" % (local_path, rsync_path))
-
-
-@task
-def database_delete():
-    """
-    Deletes the database instance
-    """
-    if(env.is_production_env is True):
-        abort(red(
-            "You are not allowed to delete the database of a production " +
-            "environment.  If this is a server restore situation, you " +
-            "have to temporarily declare env.is_production_env = False " +
-            "in the environment"))
-    execute(check_and_create_database_user)
-
-    with settings(warn_only=True), hide('stdout'):
-        checkDatabase = venvcmd('assembl-pypsql -1 -u {user} -p {password} -n {host} "{command}"'.format(
-            command="SELECT 1 FROM pg_database WHERE datname='%s'" % (env.db_database),
-            password=env.db_password, host=env.db_host, user=env.db_user))
-    if not checkDatabase.failed:
-        print(yellow("Cannot connect to database, trying to create"))
-        deleteDatabase = run('PGPASSWORD=%s dropdb --host=%s --username=%s %s' % (
-            env.db_password, env.postgres_db_host, env.db_user, env.db_database))
-        if deleteDatabase.succeeded:
-       
```

Comment:
This file should have a big-sh comment at the top as to why it exists. What Fabric does, and a brief history lesson. It needs to be explained the conditions where running it is good and well, conditions where using it crashes the system.